runtime/jcl/common/thread.cpp

```diff
@@ -603,28 +603,110 @@ Java_java_lang_Thread_setExtentLocalCache(JNIEnv *env, jclass unusedClass, jobje
 #endif /* JAVA_SPEC_VERSION >= 19 */
 
 #if JAVA_SPEC_VERSION >= 20
+/**
+ * Frame walk function to find the most recent scoped value binding in the stack.
+ *
+ * @param currentThread the current thread
+ * @param walkState the stack walk state
+ * @return J9_STACKWALK_KEEP_ITERATING or J9_STACKWALK_STOP_ITERATING
+ */
+static UDATA
+findScopedValueBindingsWalkFunction(J9VMThread *currentThread, J9StackWalkState *walkState)
+{
+	UDATA rc = J9_STACKWALK_KEEP_ITERATING;
+	if (NULL != walkState->method) {
+		J9JavaVM *vm = currentThread->javaVM;
+		J9Method *method = walkState->method;
+		J9Class *methodClass = J9_CLASS_FROM_METHOD(method);
+		J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(method);
+		J9UTF8 *methodName = J9ROMMETHOD_NAME(romMethod);
+
+		if (((J9VMJAVALANGTHREAD_OR_NULL(vm) == methodClass)
+			|| (J9VMJAVALANGVIRTUALTHREAD_OR_NULL(vm) == methodClass)
+			|| (J9VMJAVALANGSCOPEDVALUECARRIER_OR_NULL(vm) == methodClass))
+			&& J9UTF8_LITERAL_EQUALS(J9UTF8_DATA(methodName), J9UTF8_LENGTH(methodName), "runWith")
+		) {
+			if (NULL != walkState->userData1) {
+				rc = J9_STACKWALK_STOP_ITERATING;
+			} else {
+				Assert_JCL_true(0);
```

Comment:
This will assert any time we don't find the bindings. Probably best to hoist the userData1 check to the top of the function to avoid unncessary method comparisions when no binding have been found.

Reply:
Does that mean we will abandon this assertion? If we skip the method comparisons if we haven't find any bindings, we will miss the invalid case where we are in runWith method without finding a binding.

Reply:
Yes, I think the assertion would be removed.